src/olympia/amo/tests/__init__.py

```diff
@@ -204,15 +204,18 @@ def check_links(expected, elements, selected=None, verify=True):
             assert bool(e.length) == (text == selected)
 
 
-def assert_url_equal(url, other, compare_host=False):
+def assert_url_equal(url, expected, compare_host=False):
     """Compare url paths and query strings."""
     parsed = urlparse(unicode(url))
-    parsed_other = urlparse(unicode(other))
-    assert parsed.path == parsed_other.path  # Paths are equal.
-    # Params are equal.
-    assert parse_qs(parsed.query) == parse_qs(parsed_other.query)
+    parsed_expected = urlparse(unicode(expected))
+    compare_url_part(parsed.path, parsed_expected.path)
+    compare_url_part(parse_qs(parsed.query), parse_qs(parsed_expected.query))
     if compare_host:
-        assert parsed.netloc == parsed_other.netloc
+        compare_url_part(parsed.netloc, parsed_expected.netloc)
+
+
+def compare_url_part(part, expected):
+    assert part == expected, u'Expected %s, got %s' % (expected, part)
```

Comment:
tbh… doesn't pytest already provide us with the left + right details and differences in verbose mode? I'd only add a assertion message when we have more debugging information available that's not in the assertion itself.

Reply:
It does, but we never directly call this from a test, it's called inside a function that's calling another function that's parsing the content to find links which then calls the function which... by the end of it you have no idea what you are comparing. The message truly helps (the tests should be refactored to avoid doing that but... that's a lot more work)